examples/tls/tls_client.c

```diff
@@ -353,12 +360,21 @@ int TPM2_TLS_ClientArgs(void* userCtx, int argc, char *argv[])
     }
     else {
     #ifdef HAVE_ECC
-        printf("Loading ECC dummy key\n");
+        byte der[256];
+        word32 derSz = sizeof(der);
+        rc = wc_EccPublicKeyToDer(&wolfEccKey, der, derSz, 1);
+        if (rc < 0) {
+            printf("Failed to export ECC public key!\n");
+            goto exit;
+        }
+        derSz = rc;
+        rc = 0;
+
         /* Private key is on TPM and crypto dev callbacks are used */
-        /* TLS client (mutual auth) requires a dummy key loaded (workaround) */
-        if (wolfSSL_CTX_use_PrivateKey_buffer(ctx, DUMMY_ECC_KEY,
-            sizeof(DUMMY_ECC_KEY), WOLFSSL_FILETYPE_ASN1) != WOLFSSL_SUCCESS) {
-            printf("Failed to set key!\r\n");
+        /* TLS client (mutual auth) requires a public key loaded */
```

Comment:
The API is for using a private key, so this comment "LOOKS" like a mistake.  Can we add something to say that we're doing this on purpose? Something like "Even though this is API is for a private key, ....."